sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -454,11 +454,17 @@ class JacksonParser(
       schema.getFieldIndex(parser.getCurrentName) match {
         case Some(index) =>
           try {
-            row.update(index, fieldConverters(index).apply(parser))
+            val value = fieldConverters(index).apply(parser)
+            row.update(index, value)
             skipRow = structFilters.skipRow(row, index)
             bitmask(index) = false
           } catch {
             case e: SparkUpgradeException => throw e
+            case err: PartialValueException if enablePartialResults =>
+              badRecordException = badRecordException.orElse(Some(err.cause))
+              row.update(index, err.partialResult)
```

Comment:
just for my education: `err.partialResult` will be null if it's a simple column, or an array/struct/map with partial reuslt if it's array/struct/map type?

Reply:
Actually, not. When we encounter PartialValueException/`err.partialResult`, it is already a complex value, e.g. struct/map/array. If there was an error parsing a primitive value, it would be caught in NonFatal clause and then rethrown as PartialValueException.